src/Configuration/Config.AzureKeyVault/src/AzureKeyVaultConfigurationProvider.cs

```diff
@@ -78,6 +108,43 @@ private async Task LoadAsync()
             } while (secrets != null);
 
             Data = data;
+            OnReload();
+
+            if (_reloadOnChange && _pollingTask == null)
+            {
+                _pollingTask = PollForSecretChangesAsync(_reloadPollDelay, _cancellationToken.Token);
+            }
+        }
+
+        private async Task<bool> ShouldReloadAsync()
+        {
+            var secrets = await _client.GetSecretsAsync(_vault).ConfigureAwait(false);
+            if (secrets.Count() != _loadedSecrets.Count())
+            {
+                return true;
+            }
+
+            foreach (var secret in secrets)
+            {
+                if (!_loadedSecrets.ContainsKey(secret.Identifier.Name))
+                {
+                    return true;
+                }
+
+                long? currentSecretLastUpdateTick = secret.Attributes.Updated.Value.Ticks;
```

Comment:
Is there a reason to use `Updated` vs `Version`?

Reply:
We found the version field was null when the secrets were updated. The "Updated" field however, reflected the update time.